### ISSUE TITLE
[HOLD for payment 2023-04-12] [$4000] [Feature Request] Compose box should be accessible when opening a old chat

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14223]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to NewDot 
2. Open any report that you have previously chatted with, but haven’t for quite some time

## Expected Result:
The compose box is accessible and a message can be typed even while the skeleton UI loads

## Actual Result:
The compose box is blocked until the skeleton UI loads. Clicking it does nothing, even though you can see it.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.52-4
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 
![Screenshot 2023-01-09 at 07 52 34](https://user-images.githubusercontent.com/43996225/211910398-86ad2305-ba0e-40c5-b8f7-b47fec85148a.png)

https://user-images.githubusercontent.com/43996225/211910433-48d9e909-4152-487b-9b6f-b05910be0986.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @JmillsExpensify 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1673289703690979

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014ee521941ee126ec</li>
        <li>Upwork Job ID: 1617901674597654528</li>
        <li>Last Price Increase: 2023-03-28</li>
    </ul>
</details>

**Comment by techievivek** [comment]
We are already discussing this [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1673246681648109). And will get fixed in [this PR](https://github.com/Expensify/App/pull/13233).

**Comment by mallenexpensify** [comment]
Thanks @techievivek !!   Put on hold pending https://github.com/Expensify/App/issues/13088

**Comment by yuwenmemon** [comment]
@techievivek I think we're dealing with a bit of a separate issue than you dealt with in [the PR you've linked](https://github.com/Expensify/App/pull/13233). This behavior is actually happening while online as well, and the chats are still loading.

**Comment by techievivek** [comment]
@yuwenmemon Agree, it seems to be happening in both offline and online cases.

**Comment by techievivek** [comment]
> it sounds like on offline we already had a non-blocking framework in place.

We have things already in place; it's just that the compose box is disabled, so users can't take optimistic actions in both online and offline cases.

**Comment by techievivek** [comment]
Passing it to contributors to work on it.

**Comment by getusha** [comment]
### Proposal

```diff
@@ -317,7 +317,7 @@ class ReportScreen extends React.Component {
                             {(!this.isReportReadyForDisplay() || isLoadingInitialReportActions) && (
                                 <>
                                     <ReportActionsSkeletonView containerHeight={this.state.skeletonViewContainerHeight} />
-                                    <ReportFooter shouldDisableCompose isOffline={this.props.network.isOffline} />
+                                    <ReportFooter isOffline={this.props.network.isOffline} />
                                 </>
                             )}
                             <PortalHost name={CONST.REPORT.DROP_HOST_NAME} />
~
```

**Comment by sobitneupane** [comment]
@getusha Can you please elaborate your proposal? 

Your are proposing to remove `shouldDisableCompose `. In which PR was it added? What was the use of `shouldDisableCompose` then?  And Why won't it be any longer necessary?

**Comment by getusha** [comment]
### Proposal 2

```diff @@ -227,6 +227,7 @@ class ReportScreen extends React.Component {
         // (which is shown, until all the actual views of the ReportScreen have been rendered)
         const animatePlaceholder = !freeze;

+        const isReportArchived = ReportUtils.isArchivedRoom(this.props.report);
         return (
             <ScreenWrapper
                 style={screenWrapperStyle}
@@ -238,7 +239,7 @@ class ReportScreen extends React.Component {
                             <ReportHeaderSkeletonView animate={animatePlaceholder} />
                             <View style={[styles.flex1, styles.justifyContentEnd, styles.overflowHidden]}>
                                 <ReportActionsSkeletonView animate={animatePlaceholder} containerHeight={this.state.skeletonViewContainerHeig
ht} />
-                                <ReportFooter shouldDisableCompose isOffline={this.props.network.isOffline} />
+                                <ReportFooter shouldDisableCompose={false} isOffline={this.props.network.isOffline} />
                             </View>
                         </>
                     )}
@@ -300,26 +301,26 @@ class ReportScreen extends React.Component {
                                         isDrawerOpen={this.props.isDrawerOpen}
                                         parentViewHeight={this.state.skeletonViewContainerHeight}
                                     />
-                                    <ReportFooter
-                                        errors={addWorkspaceRoomOrChatErrors}
-                                        pendingAction={addWorkspaceRoomOrChatPendingAction}
-                                        isOffline={this.props.network.isOffline}
-                                        reportActions={this.props.reportActions}
-                                        report={this.props.report}
-                                        isComposerFullSize={this.props.isComposerFullSize}
-                                        onSubmitComment={this.onSubmitComment}
-                                    />
                                 </>
                             )}

                             {/* Note: The report should be allowed to mount even if the initial report actions are not loaded. If we prevent
rendering the report while they are loading then
                             we'll unnecessarily unmount the ReportActionsView which will clear the new marker lines initial state. */}
                             {(!this.isReportReadyForDisplay() || isLoadingInitialReportActions) && (
                                 <>
                                     <ReportActionsSkeletonView containerHeight={this.state.skeletonViewContainerHeight} />
-                                    <ReportFooter shouldDisableCompose isOffline={this.props.network.isOffline} />
                                 </>
                             )}
+                            {this.isReportReadyForDisplay() &&
+                                <ReportFooter
+                                    errors={addWorkspaceRoomOrChatErrors}
+                                    pendingAction={addWorkspaceRoomOrChatPendingAction}
+                                    isOffline={this.props.network.isOffline}
+                                    reportActions={this.props.reportActions}
+                                    report={this.props.report}
+                                    isComposerFullSize={this.props.isComposerFullSize}
+                                    onSubmitComment={this.onSubmitComment}
+                                />
+                            }
                             <PortalHost name={CONST.REPORT.DROP_HOST_NAME} />
                         </View>
                     </FullPageNotFoundView>
```

The Composer which is being mounted with `ReportActionsSkeletonView` is just a dummy composer and was implemented to be disabled for user interaction. 
ref: https://github.com/Expensify/App/commit/6fe2785f674a5a6933663a88dc47c02a6d336ea9
* https://github.com/Expensify/App/pull/13523
In this case have two `ReportFooter` one is with report actions and actual report state. the other one is just an empty `ReportFooter` that's non functional added to fix https://github.com/Expensify/App/pull/13523

To make the composer accessible while it's loading we can mount the functioning `ReportFooter` in general

**Comment by sobitneupane** [comment]
Thanks for the update @getusha.

Looks like the composer was intentionally disabled by [this PR](https://github.com/Expensify/App/pull/12505). So, I have [commented](https://github.com/Expensify/App/pull/12505/files#r1088840195) on the PR to know why was it disabled in the first place.

cc: @techievivek

**Comment by techievivek** [comment]
Looking at the comment [here](https://expensify.slack.com/archives/C049HHMV9SM/p1673456505880409?thread_ts=1673289703.690979&cid=C049HHMV9SM) I do agree that the code in `ReportScreen` is pretty hacky so if we are coming with a fix it has to be tested really well across all the platforms and under different scenarios.

Looking at the code changes, it makes sense to keep the `ReportFooter` component standalone once the report is ready to be displayed so users can take optimistic action irrespective of whether report actions are loaded or not.

To start with @getusha, can you please post some videos(web) with the fix for the below cases:
1. When the user is online
    -  Opening an old chat and adding a few messages to it.
2. When the user is offline
    - Opening an old chat and adding a few messages to it.
    - Opening an already opened chat and adding a few messages to it.

I will also test the changes locally and will get back to you.

**Comment by getusha** [comment]
@techievivek 

* Opening an old chat and adding a few messages to it.

https://user-images.githubusercontent.com/75031127/215466196-2a026c92-484a-4199-925b-97e13b7a5d15.mov

* Opening an old chat and adding a few messages to it. (Offline)

https://user-images.githubusercontent.com/75031127/215469637-66698ef7-b18b-40de-8693-55626054aa1f.mov

* Opening an already opened chat and adding a few messages to it.

https://user-images.githubusercontent.com/75031127/215469967-c183af9f-7d7a-417c-9de2-8be916856c9a.mov

**Comment by sobitneupane** [comment]
@getusha Can you confirm that emoji picker opens on clicking emoji button when you are offline and chat is loading? It doesn't open on my end.

**Comment by getusha** [comment]
@sobitneupane it doesn't open

**Comment by sobitneupane** [comment]
@getusha Are you working on your proposal to solve [the problem](https://github.com/Expensify/App/issues/14223#issuecomment-1408985387)? 

If anyone is working on [this issue](https://github.com/Expensify/App/issues/14223), following discussions might be helpful:
 
 https://github.com/Expensify/App/pull/12505
 https://expensify.slack.com/archives/C049HHMV9SM/p1673289703690979
 https://expensify.slack.com/archives/C01GTK53T8Q/p1669259814248689

**Comment by getusha** [comment]
The EmojiPicker is not opening because it's being rendered inside `ReportActionsView` which will not be mounted before the loading finishes.

```diff --git a/src/pages/home/ReportScreen.js b/src/pages/home/ReportScreen.js
index d544d78a65..d99bd888c5 100644
--- a/src/pages/home/ReportScreen.js
+++ b/src/pages/home/ReportScreen.js
@@ -33,6 +33,8 @@ import withLocalize from '../../components/withLocalize';
 import reportPropTypes from '../reportPropTypes';
 import FullPageNotFoundView from '../../components/BlockingViews/FullPageNotFoundView';
 import ReportHeaderSkeletonView from '../../components/ReportHeaderSkeletonView';
+import * as EmojiPickerAction from '../../libs/actions/EmojiPickerAction';
+import EmojiPicker from "../../components/EmojiPicker/EmojiPicker";

 const propTypes = {
     /** Navigation route context info provided by react navigation */
@@ -299,15 +301,6 @@ class ReportScreen extends React.Component {
                                         isDrawerOpen={this.props.isDrawerOpen}
                                         parentViewHeight={this.state.skeletonViewContainerHeight}
                                     />
-                                    <ReportFooter
-                                        errors={addWorkspaceRoomOrChatErrors}
-                                        pendingAction={addWorkspaceRoomOrChatPendingAction}
-                                        isOffline={this.props.network.isOffline}
-                                        reportActions={this.props.reportActions}
-                                        report={this.props.report}
-                                        isComposerFullSize={this.props.isComposerFullSize}
-                                        onSubmitComment={this.onSubmitComment}
-                                    />
                                 </>
                             )}

@@ -316,9 +309,20 @@ class ReportScreen extends React.Component {
                             {(!this.isReportReadyForDisplay() || isLoadingInitialReportActions) && (
                                 <>
                                     <ReportActionsSkeletonView containerHeight={this.state.skeletonViewContainerHeight} />
-                                    <ReportFooter shouldDisableCompose isOffline={this.props.network.isOffline} />
+                                    <EmojiPicker ref={EmojiPickerAction.emojiPickerRef} />
                                 </>
                             )}
+                          {this.isReportReadyForDisplay() &&
+                                <ReportFooter
+                                    errors={addWorkspaceRoomOrChatErrors}
+                                    pendingAction={addWorkspaceRoomOrChatPendingAction}
+                                    isOffline={this.props.network.isOffline}
+                                    reportActions={this.props.reportActions}
+                                    report={this.props.report}
+                                    isComposerFullSize={this.props.isComposerFullSize}
+                                    onSubmitComment={this.onSubmitComment}
+                                />
+                            }
                             <PortalHost name={CONST.REPORT.DROP_HOST_NAME} />
                         </View>
                     </FullPageNotFoundView>
```


Result


https://user-images.githubusercontent.com/75031127/215673580-93f6bdb3-8de3-4ada-8dcc-593f916fee88.mov

**Comment by techievivek** [comment]
@getusha 
Tested the changes, and it looks good(I wonder if we are missing some edge cases, though?). Also, a quick question why do we render `EmojiPicker` under `ReportActionsView` because it seems to work well even when you have imported it into the `ReportScreen` page.

**Comment by getusha** [comment]
>Also, a quick question why do we render EmojiPicker under ReportActionsView because it seems to work well even when you have imported it into the ReportScreen page.

@techievivek if i understand your question correctly we can remove the `EmojiPicker` from `ReportActionsView` and mount it under `ReportFooter` either way it works right? because as soon as the loading is completed the `EmojiPicker` from the loading unmounts and the new one mounts.

**Comment by techievivek** [comment]
@getusha Yeah, I was trying to understand can we not move `EmojiPicker` from `ReportActionsView` and mount it alongside `ReportFooter` in the `ReportScreen` page

```diff
+  {this.isReportReadyForDisplay() &&
+                                <ReportFooter
+                                    errors={addWorkspaceRoomOrChatErrors}
+                                    pendingAction={addWorkspaceRoomOrChatPendingAction}
+                                    isOffline={this.props.network.isOffline}
+                                    reportActions={this.props.reportActions}
+                                    report={this.props.report}
+                                    isComposerFullSize={this.props.isComposerFullSize}
+                                    onSubmitComment={this.onSubmitComment}
+                                />
+                                <EmojiPicker ref={EmojiPickerAction.emojiPickerRef} />
+                            }
```

or maybe under `ReportFooter` page?

**Comment by getusha** [comment]
>I was trying to understand can we not move EmojiPicker from ReportActionsView and mount it alongside ReportFooter in the ReportScreen page

Yes it makes sense to mount the EmojiPicker alongside `ReactFooter` and doesn't affect anything. do i need to update my proposal with the change?

**Comment by sobitneupane** [comment]
@getusha It looks like with the proposed changes, https://github.com/Expensify/App/issues/11856 issue will resurface.

**Comment by getusha** [comment]
@sobitneupane can you attach a video?

**Comment by sobitneupane** [comment]
@getusha I was able to reproduce it in staging as well. The Compose box is not present for fraction of seconds. So, I don't think it will be an issue.

Overall proposal looks good to me. But we must test it rigorously. Let's make sure we don't reintroduce following bugs:
https://github.com/Expensify/App/issues/11840
https://github.com/Expensify/App/issues/11856

@techievivek As [suggested](https://github.com/Expensify/App/issues/14223#issuecomment-1408505154) by @aimane-chnaif, I think we should wait for https://github.com/Expensify/App/pull/14456 to merge. Looking at the [PR](https://github.com/Expensify/App/pull/14456), some changes might be required in [the proposal](https://github.com/Expensify/App/issues/14223#issuecomment-1409781984) as well.

**Comment by techievivek** [comment]
@sobitneupane With @getusha changes can you share a video where we are re-introducing the bug of `Compose box not being present for fraction of seconds`, if that's the case then we will need a better proposal.

> As https://github.com/Expensify/App/issues/14223#issuecomment-1408505154 by @aimane-chnaif, I think we should wait for https://github.com/Expensify/App/pull/14456 to merge. Looking at the https://github.com/Expensify/App/pull/14456, some changes might be required in https://github.com/Expensify/App/issues/14223#issuecomment-1409781984 as well.

Sure.

**Comment by sobitneupane** [comment]
> @sobitneupane With @getusha changes can you share a video where we are re-introducing the bug of Compose box not being present for fraction of seconds, if that's the case then we will need a better proposal.

Sure @techievivek. It is not present for fraction of second only and is present in staging as well.

With the Proposed Changes:

https://user-images.githubusercontent.com/25876548/215979581-c4e8ea29-4d97-4c67-b5cd-85548be9520e.mov



On staging:

https://user-images.githubusercontent.com/25876548/215979615-78a5f9d8-2b62-45b5-bd32-cd05b7edf814.mov

**Comment by techievivek** [comment]
Yeah, it seems like the blank screen is present in both cases.

<img width="649" alt="Screenshot 2023-02-01 at 1 11 44 PM" src="https://user-images.githubusercontent.com/35863227/215982101-f3ce6761-7ccb-4511-b934-00af98d9df30.png">

**Comment by techievivek** [comment]
Great, thanks for the updates @aimane-chnaif. 

@JmillsExpensify Putting this on hold till we get https://github.com/Expensify/App/pull/14456 merged because we might require some changes from the PR in the [proposal](https://github.com/Expensify/App/issues/14223#issuecomment-1409781984) that @getusha has put.

**Comment by JmillsExpensify** [comment]
That sounds good, thanks!

**Comment by techievivek** [comment]
On hold, waiting for the https://github.com/Expensify/App/pull/14456 PR to get merged.

**Comment by techievivek** [comment]
Still on hold, waiting for the https://github.com/Expensify/App/pull/14456 PR to get merged.

**Comment by techievivek** [comment]
Still on hold for https://github.com/Expensify/App/pull/14456.

**Comment by JmillsExpensify** [comment]
P.S. That linked issue/PR is pretty advanced. Hopefully we'll be able to close the loop pretty shortly here. Moving this issue to weekly in the meantime.

**Comment by JmillsExpensify** [comment]
Melvin chill dude.

**Comment by mallenexpensify** [comment]
https://github.com/Expensify/App/pull/14456 is still open

**Comment by techievivek** [comment]
Not overdue, the above PR is still open.

**Comment by mallenexpensify** [comment]
PR still open https://github.com/Expensify/App/pull/14456

**Comment by JmillsExpensify** [comment]
Can't wait to get this one resolved! This continues to hit me constantly.

**Comment by sobitneupane** [comment]
@getusha Thanks for the proposal.

As [suggested](https://github.com/Expensify/App/issues/14223#issuecomment-1410443217) by @techievivek, let's move `EmojiPicker` alongside `ReportFooter`. There might be multiple edge cases that can be easily missed. So, let's make sure to test it for as many cases as possible.

🎀👀🎀 C+ reviewed

cc: @techievivek

**Comment by techievivek** [comment]
The proposal looks good to me, hiring @getusha for the job, thanks.

**Comment by MelvinBot** [comment]
📣 @getusha You have been assigned to this job by @techievivek!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by MelvinBot** [comment]
Current assignees @JmillsExpensify and @mallenexpensify are eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Current assignee @sobitneupane is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Current assignee @techievivek is eligible for the External assigner, not assigning anyone new.

**Comment by mallenexpensify** [comment]
@getusha and @sobitneupane hired!   Can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~017ad5aadeda211cef

**Comment by MelvinBot** [comment]
⚠️ Looks like this issue was linked to a **Deploy Blocker** [here](https://github.com/Expensify/App/issues/16908)

If you are the assigned CME please investigate whether the linked PR caused a regression and leave a comment with the results.

If a regression has occurred and you are the assigned CM follow the instructions [here](https://stackoverflow.com/c/expensify/questions/13647).

If this regression could have been avoided please consider also proposing a recommendation to the PR checklist so that we can avoid it in the future.

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.94-3 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/16705


If no regressions arise, payment will be issued on 2023-04-12. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by techievivek** [comment]
Hey, @mallenexpensify, it there anything else that needs to be done here, or are we good to close this one?

**Comment by mallenexpensify** [comment]
@techievivek @johnmlee101 , I see the below [here](https://github.com/Expensify/App/pull/16705#issuecomment-1496233261)
> Possible Regression: https://github.com/Expensify/App/issues/16908

Do either of you have any further insight or details before I pay? 

Also.. for payment, below is the bonus timeline. 

Hired - March 29
<img width="377" alt="image" src="https://user-images.githubusercontent.com/22508304/231898964-7fb707bb-a40d-4fa0-8d8b-d05571953b98.png">

Merged March 31
<img width="268" alt="image" src="https://user-images.githubusercontent.com/22508304/231898971-588cb352-50df-4b8a-bc32-6bba534fa685.png">

**Comment by techievivek** [comment]
@mallenexpensify Yes, there was a regression from the original PR here https://github.com/Expensify/App/issues/16908 which was also fixed @getusha

**Comment by mallenexpensify** [comment]
Thanks @techievivek and @getusha. [Per CONTRIBUTING.md.](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#:~:text=If%20the%20PR%20causes%20a%20regression%20within%207%20days%20of%20being%20deployed%20to%20production%2C%20contributors%20are%20not%20eligible%20for%20the%2050%25%20bonus.) 
> If the PR causes a regression within 7 days of being deployed to production, contributors are not eligible for the 50% bonus.

So... if the regression was before production, that shouldn't affect the bonus for @getusha . 

@sobitneupane , this is from the C+ process doc
> If regressions are found that “should have”* been caught after the PR has been approved, the C+’s payment gets cut in half for each regression found (½ the first time, ½  the second, etc (ie. $1000 > $500 > $250)). This includes regressions on staging and production. Idea discussion slack [here](https://expensify.slack.com/archives/C01SKUP7QR0/p1631151330241000). Any bonus increases that might have been added for timeliness are rescinded for regressions. 
*“Should have” = C+ should have caught the bug by fully following the checklist.  If C+ skips a step or completed the checklist incompletely, payment will be cut in half.

If you don't think you "should have" caught the regression, can you provide reasoning?

**Comment by sobitneupane** [comment]
> If you don't think you "should have" caught the regression, can you provide reasoning?

I think I should have caught it earlier though it is a edge case.

**Comment by techievivek** [comment]
Not overdue, we are just finalizing the payment here.

**Comment by mallenexpensify** [comment]
@aimane-chnaif , checking on here https://expensify.slack.com/archives/C02NK2DQWUX/p1681749639671849

> So... if the regression was before production, that shouldn't affect the bonus for @getusha .

Does that mean that if regression found on main or staging, bonus still applies to contributor?

**Comment by mallenexpensify** [comment]
Thanks for the ping @getusha , we're in the process of updating the language for CONTRIBUTING.md to denote that any regression cancels the bonus because there's no benefit for timeliness if there are regressions that require additional resources to address.

> If the PR causes a regression within 7 days of being deployed to production, contributors are not eligible for the 50% bonus.

`within 7 days of being deployed to production` is the part we still need to _honor_ here and the regression was not within that time period, it was before, so the bonus stands for now.  

@sobitneupane paid $2000 and @getusha paid $6000.  Anticipate CONTRIBUTING.md being updated very soon with the new language stating that timeliness bonuses won't be eligible if there are regressions.

**Comment by mallenexpensify** [comment]
For some reason this didn't get posted, @sobitneupane , can you complete the below then I'll create a GH to update TestRail?

**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@sobitneupane] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@sobitneupane] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@sobitneupane] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@mallenexpensify] Determine if we should create a regression test for this bug.
- [x] [@sobitneupane] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@mallenexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [x]  [@sobitneupane] The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/12505
> * [x]  [@sobitneupane] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/12505/files#r1088840195

> * [x]  [@sobitneupane] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

It is an improvement in the app rather than bug fix as discussed earlier in https://github.com/Expensify/App/pull/12505/files#r1088840195

**Comment by sobitneupane** [comment]
**Regression Test Proposal**

1. Open Expensify app
2. Disconnect the device from the internet
3. Open any old chat or a chat that is still loading
4. Type a message in the compose box.
5. Open EmojiPicker and insert an emoji
6. Send the message by clicking the send icon
7. Add any attachment and send the attachment
9. Confirm Sending, splitting, requesting money works as expected
10. Verify that the compose box is accessible and functioning properly(i.e, you are able to perform step 3, step 4, step 5, step 6, and step 7) as it would when the device is connected to the internet

Do we agree 👍 or 👎

**Comment by JmillsExpensify** [comment]
Thank you so much for getting this one out and in production! It's really great. ❤️

**Comment by mallenexpensify** [comment]
Thanks @sobitneupane (and @getusha for the 👍)
GH created to update TestRail with regression test steps
https://github.com/Expensify/Expensify/issues/277753
